src/components/SiteAlert/SiteAlert.test.tsx

```diff
@@ -0,0 +1,124 @@
+/*  eslint-disable jsx-a11y/anchor-is-valid, react/jsx-key */
+import React from 'react'
+import { render } from '@testing-library/react'
+
+import { SiteAlert } from './SiteAlert'
+
+const testChildren = (
+  <p className="usa-alert__text">
+    some default text&nbsp;
+    <a className="usa-link" href="#">
+      with a link
+    </a>
+    .
+  </p>
+)
+
+const testDefaultProps = {
+  heading: 'test heading',
+  children: testChildren,
+}
+
+const testChildrenWithList = (
+  <ul>
+    <li>
+      some default text&nbsp;
+      <a className="usa-link" href="#">
+        with a link&nbsp;
+      </a>
+    </li>
+    <li>
+      another list item&nbsp;
+      <a className="usa-link" href="#">
+        with a link
+      </a>
+    </li>
+    <li>another list item, no link</li>
+  </ul>
+)
+
+describe('SiteAlert component', () => {
+  it('renders without errors', () => {
+    const { getByTestId, getByRole } = render(
+      <SiteAlert variant="info" {...testDefaultProps} />
+    )
+
+    expect(getByTestId('siteAlert')).toBeInTheDocument()
+    expect(getByRole('link')).toBeInTheDocument()
+  })
+
+  it('accepts a className', () => {
+    const { getByTestId } = render(
+      <SiteAlert
+        variant="info"
+        className="custom-class-name"
+        {...testDefaultProps}
+      />
+    )
+
+    expect(getByTestId('siteAlert')).toHaveClass(
+      'usa-site-alert usa-site-alert--info custom-class-name'
+    )
+  })
+
+  it('accepts a custom aria-label', () => {
+    const { getByTestId } = render(
+      <SiteAlert
+        variant="emergency"
+        aria-label="custom aria label"
+        {...testDefaultProps}
+      />
+    )
+
+    expect(getByTestId('siteAlert')).toHaveAttribute(
+      'aria-label',
+      'custom aria label'
+    )
+  })
+
+  it('renders a passed in heading', () => {
+    const { getByRole } = render(
+      <SiteAlert variant="info" {...testDefaultProps} />
+    )
+
+    expect(getByRole('heading')).toBeInTheDocument()
+    expect(getByRole('heading')).toHaveClass('usa-alert__heading')
```

Comment:
ooo and a quick follow up on these lines since I didn't notice on my first pass:

if you ever find yourself doing `getBy___` or `queryBy___` for the same thing multiple times in a test you can factor that out into a const, e.g:

```TSX
const heading = getByRole('heading')
expect(heading).toBeInTheDocument()
expect(heading).toHaveClass('usa-alert__heading')
```

I'm not sure if the performance impact of querying the dom for a specific element is high, but its an easy optimization we can do to keep the tests efficient.